### ISSUE TITLE
extra credit on hw3 for using automatic test case tool

### ISSUE THREAD

**Issue by mikeizbicki** [mikeizbicki/ucr-cs100#1064]
@M-Evans recently added (#1063) a tool to help automate the task of testing your rshell program.  You can find instructions on how to use it [here](https://github.com/mikeizbicki/ucr-cs100/tree/2015winter/scripts/cs100-runtests).  The command `cs100-runtests` is now in your `PATH` on hammer, so you don't have to do any installation to use the tool.

I am offering 5 pts of extra credit to anyone who:
1. uses this tool to test their rshell hw3 submission.
2. provides detailed feedback on how this tool was useful and/or can be improved; you should reply to this issue to provide the feedback.

**Comment by tristanzickovich** [comment]
Major things I noticed about this tool:

**Pros:**
- Being able to type test cases into a file before scripting allows for easy organization of test cases
- Forgotten test cases can be added easily
- Generating  a new script (for forgotten test cases) is painless compared to the old method
- It enhances legibility of script file. It eliminates all of the backspace and messed up typed characters that appear in normal scripting files.

**Cons:**
- The default speed is a bit too quick.  It should be slowed a little to allow time for more time consuming commands, such as compiling. Some commands are skipped if they take too long, messing with the rest of the test cases.
- In the README, I feel you should add a section explaining briefly how to use and run your script.  Being on HW3, I initially skipped over the HW0 section where this was explained.  It should have its own section for quick reference right above the first HW section.

Overall it is a very useful tool.

**Comment by Fire3galaxy** [comment]
Pro: Super easy to run, and being able to go between bash and rshell is major plus!
Con: Current path name is usually before prompt, but outputs after prompt for bash. When I do the style of 
`#Comment here.`
`cd ../`
It takes 4 lines to show this

**Comment by MiaoXiao** [comment]
Since we have to have ^C on its line, does that mean we can't test cases that include ^C and something like ls? For example, if I was trying to test

`ls ^C`

It does not seem like it will work. If this is actually the case, that means there are some test cases you can't test with HW3 using this tool.

Those are my thoughts so far, although I will probably have more to say if/when I get the tool to work correctly; I am not sure if it's a problem on my end or with the tool, but my command prompt keeps endlessly looping when I use the tool, although it works fine with BASH.

**Comment by M-Evans** [comment]
I liked using it. :smile: 

suggested improvement:
make the shell the last parameter specified. After the shell parameter, all other parameters would be assumed parameters to the shell. This would extend the ability of `cs100-runtests` to test the behavior of shells under more conditions than those without any parameters.

**Comment by NisreenFarhoud** [comment]
Pros: 
- helps user organize test cases
- makes file more readable

Cons:
- took awhile to understand how to use the tool
- more detailed explanation of the second argument would be helpful (why in some examples its 'bin/rshell' and others its 'rshell'

**Comment by yliu075** [comment]
This tool provides a fast and easy way to test the functionality of rshell. It improves the readability of the test scripts. The time parameter is useful to increase to accuracy of the tests. 
But there is a bug I found: when I use `>` to redirect the output to a file, it only works when there are no error with the test cases, but on edge cases with perror outputs only 3 lines get redirected:

```
exit

================================================================================
```

**Comment by AndyThio** [comment]
Pros:
- Making the script for testcases becomes extremely easy
- adding testcases becomes easier

Cons:
- Using Ctrl+C with a high delay on something like cat, printed out my terminal in an endless loops and the fix wasn't very clear.
- Including a general use section in your README

**Comment by Anthony1234567** [comment]
This saved me a bunch of time though for some reason when running the test script. the harness defaults to your HOME directory and not the current directory you were working on before starting the test. Having to account for that, This still is faster than typing up the script after running the script command in Bash.

**Comment by GitHubbubble** [comment]
This program really helps me organize my test cases and reveiwing cases as a whole means I avoid costly errors that may occur from checking one at a time ( I make a change that actually change the result of a previous test). So pretty decent time saver!

I encountered some odd errors when running the example script for hw0. My rshell had an issue where it began looping indefinetly. This was very dangerous as by now I had handled ^C. If there's any way to implement an easy kill function in the case of a serious issue, it may be worth your while to do so!
